src/app/qgsmergeattributesdialog.cpp

```diff
@@ -460,21 +484,49 @@ void QgsMergeAttributesDialog::mFromSelectedPushButton_clicked()
     return;
   }
 
-  for ( int i = 0; i < mTableWidget->columnCount(); ++i )
-  {
-    QComboBox *currentComboBox = qobject_cast<QComboBox *>( mTableWidget->cellWidget( 0, i ) );
-    if ( !currentComboBox )
-      continue;
+  setAllAttributesFromFeature( featureId );
+}
 
-    if ( mVectorLayer->fields().at( i ).constraints().constraints() & QgsFieldConstraints::ConstraintUnique )
+void QgsMergeAttributesDialog::mFromLargestPushButton_clicked()
+{
+  QgsFeatureId featureId;
+  double maxValue = 0;
+
+  switch ( mVectorLayer->geometryType() )
+  {
+    case QgsWkbTypes::LineGeometry:
     {
-      currentComboBox->setCurrentIndex( currentComboBox->findData( QStringLiteral( "skip" ) ) );
+      QgsFeatureList::const_iterator f_it = mFeatureList.constBegin();
+      for ( ; f_it != mFeatureList.constEnd(); ++f_it )
+      {
+        double featureLength = f_it->geometry().length();
+        if ( featureLength > maxValue )
```

Comment:
if all lengths are zero then featureId will never be set. Possibly a dumb corner case, but maybe this should be >= or maxValue should start at -1 ?

Reply:
This is kind of on purpose. There is no reason to select one featureId randomly when there is not one larger geometry, and there is line 528 to safeguard uninitialized use of the featureId.

Reply:
Ah - gotcha. All good then!